<a href="https://colab.research.google.com/github/SandraMangas/AnalliticaDatos_ProyectoSucre/blob/main/Proyecto.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Misión 2 - Proyecto de Analítica de Datos
    Integrantes:
          - Omar Rojas Mulett
          - Álvaro Galvis García
          - Luis Alberto Pérez Oviedo
          - Sandra Patricia Mangas Galván

*Bootcamp Análisis de Datos - Talento Tech MinTIC - Julio de 2025*.

Identificación del Problema

En el departamento de Sucre, muchas instituciones educativas presentan limitaciones en su acceso a servicios de conectividad (internet, plataformas digitales, recursos tecnológicos). Estas condiciones podrían estar influyendo en los resultados académicos de los estudiantes, especialmente en pruebas estandarizadas como las Saber 11, que son clave para el ingreso a la educación superior.
¿Existe una correlación significativa entre el nivel de conectividad institucional (acceso a internet y uso de tecnologías) y los puntajes obtenidos en las pruebas Saber 11 en el departamento de Sucre?

## 1. Importar Librerias

In [46]:
import pandas as pd  # Para manipulación y análisis de datos.
import numpy as np # Para operaciones numéricas.
import matplotlib.pyplot as plt # Para crear visualizaciones estáticas.
import seaborn as sns # Para crear gráficos estadísticos.
import requests # Trae datos de un sitio web o API.

### 2.1 Intituciones Educativas

## 2. Cargar Datos

In [47]:
# URL de la API OData
url_institucionesE = "https://www.datos.gov.co/api/odata/v4/j7te-zasu"

# Solicitud a la API
response = requests.get(url_institucionesE) # requests.get() obtiene los datos JSON desde la API.obtiene los datos JSON desde la API.
data_url_institucionesE = response.json()

# Convertir los datos de 'value' a DataFrame legible
df_institucionesE = pd.json_normalize(data_url_institucionesE['value']) # data['value'] accede a la lista de registros reales.

### 2.2 Conectividad

In [48]:
# URL de la API OData
url_conectividad = "https://www.datos.gov.co/api/odata/v4/a2f2-cnws"

# Solicitud a la API
response = requests.get(url_conectividad) # requests.get() obtiene los datos JSON desde la API.obtiene los datos JSON desde la API.
data_url_conectividad = response.json()

# Convertir los datos de 'value' a DataFrame
df_conectividad = pd.json_normalize(data_url_conectividad['value']) # data['value'] accede a la lista de registros reales.

### 2.3 Pruebas Saber 11

#### 2.3.1 Dataset de Datos Abiertos

In [49]:
# URL la API con extensión JSON y límite
url_saber11 = "https://www.datos.gov.co/resource/kgxf-xxbe.json?$limit=50000"

# Solicitud a la API
response = requests.get(url_saber11)
data_url_saber11 = response.json()

# Convertir directamente a DataFrame
df_saber11 = pd.DataFrame(data_url_saber11)

# Nota importante: Si se usa .json, no se requiere .odata ni .api.
# https://www.datos.gov.co/api/odata/v4/kgxf-xxbe

In [50]:
print(df_saber11.columns)

Index(['periodo', 'estu_tipodocumento', 'estu_consecutivo',
       'cole_area_ubicacion', 'cole_bilingue', 'cole_calendario',
       'cole_caracter', 'cole_cod_dane_establecimiento', 'cole_cod_dane_sede',
       'cole_cod_depto_ubicacion', 'cole_cod_mcpio_ubicacion',
       'cole_codigo_icfes', 'cole_depto_ubicacion', 'cole_genero',
       'cole_jornada', 'cole_mcpio_ubicacion', 'cole_naturaleza',
       'cole_nombre_establecimiento', 'cole_nombre_sede',
       'cole_sede_principal', 'estu_cod_depto_presentacion',
       'estu_cod_mcpio_presentacion', 'estu_cod_reside_depto',
       'estu_cod_reside_mcpio', 'estu_depto_presentacion', 'estu_depto_reside',
       'estu_estadoinvestigacion', 'estu_estudiante', 'estu_fechanacimiento',
       'estu_genero', 'estu_mcpio_presentacion', 'estu_mcpio_reside',
       'estu_nacionalidad', 'estu_pais_reside', 'estu_privado_libertad',
       'fami_cuartoshogar', 'fami_educacionmadre', 'fami_educacionpadre',
       'fami_estratovivienda', 'fami_perso

In [51]:
df_saber11["cole_depto_ubicacion"] = df_saber11["cole_depto_ubicacion"].str.strip().str.upper()
# Filtrar solo las filas donde el departamento sea SUCRE
# Asegúrate que el nombre esté escrito exactamente así:
df_saber11_sucre = df_saber11[df_saber11["cole_depto_ubicacion"] == "SUCRE"]

#### 2.3.2 Dataset de Repositorio ICFES

In [52]:
# Cargar el dataset 'PruebasSaber11.csv' desde GitHub.
# Este archivo fue transformado previamente con Power Query: se eliminaron columnas irrelevantes
# y se filtraron los datos del departamento de Sucre.

url_saber11_icfes = "https://raw.githubusercontent.com/SandraMangas/AnalliticaDatos_ProyectoSucre/main/PruebasSaber11.csv"
df_saber11_icfes = pd.read_csv(url_saber11_icfes, encoding='latin1', sep=';')

## 3. Exploración Inicial

### 3.1 Intituciones Educativas

In [53]:
df_institucionesE.head(20)

__id         nombre_establecimiento  codigo_dane_sede  \
0   row-ey8z~kj7g~u8nv           INST EDUC BUENAVISTA      170110000025   
1   row-ejkd.f47a~qiua           INST EDUC BUENAVISTA      170110000033   
2   row-inth.2vhr_qrty           INST EDUC BUENAVISTA      170110000041   
3   row-f5bn.cir7~hizu           INST EDUC BUENAVISTA      170110000050   
4   row-nxx4_bjci-bsqm    INST EDUC SAN JUAN BAUTISTA      170124000001   
5   row-nr9f_qnjr-wzdk    INST EDUC SAN JUAN BAUTISTA      170124000010   
6   row-85zw~z36b_4csg    INST EDUC SAN JUAN BAUTISTA      170124000192   
7   row-k25g-yhmr.4ig3    INST EDUC SAN JUAN BAUTISTA      170124000265   
8   row-mjvn_sigp-hfb3            I.E. VICTOR ZUBIRIA      170204000053   
9   row-876m.pia4~paax            I.E. VICTOR ZUBIRIA      170204000061   
10  row-ejpc-h8bg_9ncj            I.E. VICTOR ZUBIRIA      170204000096   
11  row-5wra-qjn9-8us7            I.E. VICTOR ZUBIRIA      270204000023   
12  row-dzc5~8uzh_pmsc            I.E. VICTOR ZUBIRIA      270204000104   
13  row-q24j-szek~2u2d            I.E. VICTOR ZUBIRIA      270204000163   
14  row-yeje.ajn6~acfi              INST EDUC PIO XII      170215000047   
15  row-xppm-kb55_e5tb              INST EDUC PIO XII      170215000080   
16  row-gq7w.xprk_xtmb              INST EDUC PIO XII      170215013645   
17  row-b5g8-6qvs.utkp  I.E. FRANCISCO JOSE DE CALDAS      170215000021   
18  row-c2aj~gz6j.xnx5  I.E. FRANCISCO JOSE DE CALDAS      170215000055   
19  row-ydrx~vmwy~kub4  I.E. FRANCISCO JOSE DE CALDAS      170215001027   

                                       nombre_sede    zona  
0                             INST EDUC BUENAVISTA  Urbano  
1                  SEDE NUESTRA SEÑORA DEL ROSARIO  Urbano  
2                       SEDE SAN FRANCISCO DE ASIS  Urbano  
3                           SEDE MARIA AUXILIADORA  Urbano  
4                                 SEDE ROSA GARCIA  Urbano  
5                          SEDE SAN VICENTE FERRER  Urbano  
6               SEDE SANTA TERESITA DEL NIÑO JESUS  Urbano  
7                      INST EDUC SAN JUAN BAUTISTA  Urbano  
8             I.E. VICTOR ZUBIRIA - SEDE PRINCIPAL  Urbano  
9                             SEDE URBANA DE NIÑAS  Urbano  
10                             SEDE LAS CAMPESINAS  Urbano  
11                                   C.E. EL CERRO   Rural  
12                                       CHINULITO   Rural  
13                                           CEIBA   Rural  
14                               INST EDUC PIO XII  Urbano  
15                           SEDE HIJOS DEL CHOFER  Urbano  
16                            SEDE EL CARABINERITO  Urbano  
17                     SEDE MONSENOR CARMELO PERCY  Urbano  
18  I.E. FRANCISCO JOSE DE CALDAS - SEDE PRINCIPAL  Urbano  
19                              SEDE TRES DE MARZO  Urbano

In [54]:
df_institucionesE.tail(20)

__id                             nombre_establecimiento  \
780  row-hjgm-336m-ht5p                                I.E. SANTA TERESITA   
781  row-sgc2.ibvg_uwqk                                      I.E. EL MAMON   
782  row-uvnc~2cic.k8zg                                      I.E. EL MAMON   
783  row-kfhc.q2u9~4kfq                                      I.E. EL MAMON   
784  row-7sbp_gbe4~tzhi                                      I.E. EL MAMON   
785  row-tu6f_3mng-34p4                       CENTRO EDUCATIVO LOS COROZOS   
786  row-5f2s-9nv3-fwmd                       CENTRO EDUCATIVO LOS COROZOS   
787  row-46ha.spmb.hh38                       CENTRO EDUCATIVO LOS COROZOS   
788  row-mfvp~5vnq-z45g                       CENTRO EDUCATIVO LOS COROZOS   
789  row-ejjj~gawc~mjqy                                     I.E. PALO ALTO   
790  row-5hxx.bi22-ftin                                     I.E. PALO ALTO   
791  row-pm2m~65af_66hx                                     I.E. PALO ALTO   
792  row-jhhp-wu7e_kbim                                     I.E. PALO ALTO   
793  row-cx6n~rcrs-q88c                                     I.E. PALO ALTO   
794  row-7y8t~7qmj_cj7c                                     I.E. PALO ALTO   
795  row-ua8v-t686_m7wh                                     I.E. PALO ALTO   
796  row-7pap~isad~c28q  I.E. TÉCNICO AGROPECUARIO CARLOS ARTURO VERBEL...   
797  row-tm6h.uxeq-f65n  I.E. TÉCNICO AGROPECUARIO CARLOS ARTURO VERBEL...   
798  row-zbui.qah8~9ixi                     I.E. LA  INMACULADA CONCEPCION   
799  row-iri7_ud8c~xekj                     I.E. LA  INMACULADA CONCEPCION   

     codigo_dane_sede                                        nombre_sede  \
780      370820000112               I.E. SANTA TERESITA - SEDE PRINCIPAL   
781      270215000297                                         CANTAGALLO   
782      270215000866                                          LAS TINAS   
783      270215001293                               RINCON DE LAS FLORES   
784      470215001110                     I.E. EL MAMON - SEDE PRINCIPAL   
785      270429000847    CENTRO EDUCATIVO SANTISIMA CRUZ DE HUIRA ARRIBA   
786      270429001533                      CENTRO EDUCATIVO LA MILAGROSA   
787      270429002220              CENTRO EDUCATIVO SAN JOSE HUIRA ABAJO   
788      470429001966                       CENTRO EDUCATIVO LOS COROZOS   
789      170713000879                                 PITA CAPACHO NO. 2   
790      270713000016                            PREESCOLAR DE PALO ALTO   
791      270713000172                              PRIMARIA DE PALO ALTO   
792      270713000521                                BOCAS DE MARIA NO.1   
793      270713000741                                           LA UNION   
794      270713000911                                             BOLITO   
795      470713001062                    I.E. PALO ALTO - SEDE PRINCIPAL   
796      270823000224                         CENTRO EDUCATIVO LOS ALTOS   
797      470823000461  I.E. TÉCNICO AGROPECUARIO CARLOS ARTURO VERBEL...   
798      270823005196                      CENTRO EDUCATIVO LA ESPERANZA   
799      470823000479     I.E. LA INMACULADA CONCEPCION - SEDE PRINCIPAL   

       zona  
780  Urbano  
781   Rural  
782   Rural  
783   Rural  
784   Rural  
785   Rural  
786   Rural  
787   Rural  
788   Rural  
789   Rural  
790   Rural  
791   Rural  
792   Rural  
793   Rural  
794   Rural  
795   Rural  
796   Rural  
797   Rural  
798   Rural  
799   Rural

In [55]:
df_institucionesE.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 800 entries, 0 to 799
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   __id                    800 non-null    object
 1   nombre_establecimiento  800 non-null    object
 2   codigo_dane_sede        800 non-null    int64 
 3   nombre_sede             800 non-null    object
 4   zona                    800 non-null    object
dtypes: int64(1), object(4)
memory usage: 31.4+ KB


### 3.2 Conectividad

In [56]:
df_conectividad.head()

__id nombre_departamento nombre_municipio pdet  \
0  row-x3fv_7fq4.tx3x               SUCRE         MAJAGUAL   NO   
1  row-5rbh.85xj_2mc9               SUCRE           OVEJAS   SI   
2  row-vpvn_jhri.m68b               SUCRE           OVEJAS   SI   
3  row-65em.fubm-dvu4               SUCRE       TOLÚ VIEJO   SI   
4  row-er2d.32h4.7fje               SUCRE       TOLÚ VIEJO   SI   

   nombre_centro_poblado                              nombre_sede_educativa  \
0      CORREG EL NARANJO                               INST EDUC EL NARANJO   
1         VEREDA DAMASCO                           CENTRO EDUCATIVO DAMASCO   
2  VEREDA VILLA COLOMBIA                    CENTRO EDUCATIVO VILLA COLOMBIA   
3  CORREG DE LAS PIEDRAS  I.E. TÉCNICO AGROPECUARIO DE LAS PIEDRAS - SED...   
4      CORREG LA PALMIRA                      INSTITUCION EDUCATIVA PALMIRA   

  fecha_inicio_de_operaci_n fin_de_operaci_n_inicial       estado  
0                      None                     None  INSTALACIÓN  
1   2021-09-25T00:00:00.000  2031-09-25T00:00:00.000    OPERACIÓN  
2   2021-09-25T00:00:00.000  2031-09-25T00:00:00.000    OPERACIÓN  
3                      None                     None  INSTALACIÓN  
4                      None                     None  INSTALACIÓN

In [57]:
df_conectividad.tail()

__id nombre_departamento nombre_municipio pdet  \
470  row-639p-mi9q_cjp7               SUCRE         GUARANDA   NO   
471  row-ium6~g7ec.8qki               SUCRE         GUARANDA   NO   
472  row-hsbd_3sav~txc2               SUCRE           CHALÁN   SI   
473  row-bu3d.kbb2_jegg               SUCRE          COROZAL   NO   
474  row-ffu2.rtx6~vzda               SUCRE           COLOSÓ   SI   

    nombre_centro_poblado nombre_sede_educativa fecha_inicio_de_operaci_n  \
470     VDA. LAS MERCEDES          LAS MERCEDES                      None   
471     VDA. PUEBLO NUEVO          PUEBLO NUEVO                      None   
472         DESBARRANCADO         LOS COMUNEROS                      None   
473        VEREDA LA LOMA               LA LOMA                      None   
474          CORREG CEIBA                 CEIBA   2022-01-01T00:00:00.000   

    fin_de_operaci_n_inicial         estado  
470                     None  EN PLANEACIÓN  
471                     None    INSTALACIÓN  
472                     None  EN PLANEACIÓN  
473                     None    INSTALACIÓN  
474  2032-01-01T00:00:00.000      OPERACIÓN

In [58]:
df_conectividad.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 475 entries, 0 to 474
Data columns (total 9 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   __id                       475 non-null    object
 1   nombre_departamento        475 non-null    object
 2   nombre_municipio           475 non-null    object
 3   pdet                       475 non-null    object
 4   nombre_centro_poblado      475 non-null    object
 5   nombre_sede_educativa      475 non-null    object
 6   fecha_inicio_de_operaci_n  179 non-null    object
 7   fin_de_operaci_n_inicial   179 non-null    object
 8   estado                     475 non-null    object
dtypes: object(9)
memory usage: 33.5+ KB


### 3.3 Saber 11

#### 3.3.1 Dataset de Datos Abiertos

In [59]:
df_saber11.head()

periodo estu_tipodocumento  estu_consecutivo cole_area_ubicacion  \
0   20131                 CR  SB11201310000414              URBANO   
1   20194                 TI  SB11201940464873               RURAL   
2   20194                 TI  SB11201940464873               RURAL   
3   20122                 TI  SB11201220204399              URBANO   
4   20132                 TI  SB11201320464198              URBANO   

  cole_bilingue cole_calendario      cole_caracter  \
0             N               B          ACADÉMICO   
1             N               A  TÉCNICO/ACADÉMICO   
2             N               A  TÉCNICO/ACADÉMICO   
3             N               A  TÉCNICO/ACADÉMICO   
4             N               A            TÉCNICO   

  cole_cod_dane_establecimiento cole_cod_dane_sede cole_cod_depto_ubicacion  \
0                  311848000812       311848000812                       11   
1                  141016000305       241016000342                       41   
2                  141016000305       241016000342                       41   
3                  163130000718       163130000718                       63   
4                  119001000559       119001000559                       19   

   ... fami_tienecomputador fami_tieneinternet fami_tienelavadora  \
0  ...                   Si                 Si                 Si   
1  ...                   Si                 Si                 Si   
2  ...                   Si                 Si                 Si   
3  ...                   Si                 Si                 No   
4  ...                   Si                 Si                 Si   

  desemp_ingles punt_ingles punt_matematicas punt_sociales_ciudadanas  \
0            B+          94               88                      NaN   
1            B1          71               66                       70   
2            B1          71               66                       70   
3            A1          48               45                      NaN   
4            A-          43               52                      NaN   

  punt_c_naturales punt_lectura_critica punt_global  
0              NaN                  NaN         NaN  
1               65                   69         339  
2               65                   69         339  
3              NaN                  NaN         NaN  
4              NaN                  NaN         NaN  

[5 rows x 51 columns]

In [60]:
df_saber11.tail()

periodo estu_tipodocumento  estu_consecutivo cole_area_ubicacion  \
49995   20194                 TI  SB11201940623464               RURAL   
49996   20132                 TI  SB11201320493041              URBANO   
49997   20112                 TI  SB11201120318868              URBANO   
49998   20131                 TI  SB11201310079267              URBANO   
49999   20132                 TI  SB11201320070946              URBANO   

      cole_bilingue cole_calendario      cole_caracter  \
49995           NaN               A                NaN   
49996           NaN               A          ACADÉMICO   
49997             N               A          ACADÉMICO   
49998             N               A  TÉCNICO/ACADÉMICO   
49999             N               A          ACADÉMICO   

      cole_cod_dane_establecimiento cole_cod_dane_sede  \
49995                  418001800021       418001800021   
49996                  373411002242       373411002242   
49997                  311001011800       311001011800   
49998                  386568061015       386568061015   
49999                  115001000057       115001000057   

      cole_cod_depto_ubicacion  ... fami_tienecomputador fami_tieneinternet  \
49995                       18  ...                   Si                 Si   
49996                       73  ...                   Si                 No   
49997                       11  ...                   Si                 Si   
49998                       86  ...                   No                 No   
49999                       15  ...                   No                 No   

      fami_tienelavadora desemp_ingles punt_ingles punt_matematicas  \
49995                 Si            A-          47               45   
49996                 Si            A1          51               49   
49997                 Si            A1          49               52   
49998                 No            A-          27               44   
49999                 No            A2          59               74   

      punt_sociales_ciudadanas punt_c_naturales punt_lectura_critica  \
49995                       45               45                   54   
49996                      NaN              NaN                  NaN   
49997                      NaN              NaN                  NaN   
49998                      NaN              NaN                  NaN   
49999                      NaN              NaN                  NaN   

      punt_global  
49995         236  
49996         NaN  
49997         NaN  
49998         NaN  
49999         NaN  

[5 rows x 51 columns]

In [61]:
df_saber11.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 51 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   periodo                        50000 non-null  object
 1   estu_tipodocumento             50000 non-null  object
 2   estu_consecutivo               50000 non-null  object
 3   cole_area_ubicacion            49977 non-null  object
 4   cole_bilingue                  44133 non-null  object
 5   cole_calendario                49987 non-null  object
 6   cole_caracter                  49460 non-null  object
 7   cole_cod_dane_establecimiento  49977 non-null  object
 8   cole_cod_dane_sede             49997 non-null  object
 9   cole_cod_depto_ubicacion       49998 non-null  object
 10  cole_cod_mcpio_ubicacion       49998 non-null  object
 11  cole_codigo_icfes              49809 non-null  object
 12  cole_depto_ubicacion           49998 non-null  object
 13  c

In [62]:
df_saber11_sucre.info()

<class 'pandas.core.frame.DataFrame'>
Index: 907 entries, 21 to 49925
Data columns (total 51 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   periodo                        907 non-null    object
 1   estu_tipodocumento             907 non-null    object
 2   estu_consecutivo               907 non-null    object
 3   cole_area_ubicacion            907 non-null    object
 4   cole_bilingue                  778 non-null    object
 5   cole_calendario                907 non-null    object
 6   cole_caracter                  906 non-null    object
 7   cole_cod_dane_establecimiento  907 non-null    object
 8   cole_cod_dane_sede             907 non-null    object
 9   cole_cod_depto_ubicacion       907 non-null    object
 10  cole_cod_mcpio_ubicacion       907 non-null    object
 11  cole_codigo_icfes              905 non-null    object
 12  cole_depto_ubicacion           907 non-null    object
 13  cole_ge

### 3.3.2 Dataset de Repositorio ICFES

In [63]:
df_saber11_icfes.head()

periodo cole_area_ubicacion  cole_cod_dane_establecimiento  \
0    20241              URBANO                   370670800018   
1    20241              URBANO                   370670800018   
2    20241              URBANO                   370670800018   
3    20241              URBANO                   370670800018   
4    20241              URBANO                   370670800018   

   cole_cod_dane_sede cole_depto_ubicacion cole_jornada cole_mcpio_ubicacion  \
0        370670800018                SUCRE     SABATINA              SAMPUÉS   
1        370670800018                SUCRE     SABATINA              SAMPUÉS   
2        370670800018                SUCRE     SABATINA              SAMPUÉS   
3        370670800018                SUCRE     SABATINA              SAMPUÉS   
4        370670800018                SUCRE     SABATINA              SAMPUÉS   

  cole_naturaleza                        cole_nombre_establecimiento  \
0      NO OFICIAL  INSTITUCION EDUCATIVA GIMNASIO PEDAGOGICO SAMP...   
1      NO OFICIAL  INSTITUCION EDUCATIVA GIMNASIO PEDAGOGICO SAMP...   
2      NO OFICIAL  INSTITUCION EDUCATIVA GIMNASIO PEDAGOGICO SAMP...   
3      NO OFICIAL  INSTITUCION EDUCATIVA GIMNASIO PEDAGOGICO SAMP...   
4      NO OFICIAL  INSTITUCION EDUCATIVA GIMNASIO PEDAGOGICO SAMP...   

                                    cole_nombre_sede  ...  \
0  INSTITUCION EDUCATIVA GIMNASIO PEDAGOGICO SAMP...  ...   
1  INSTITUCION EDUCATIVA GIMNASIO PEDAGOGICO SAMP...  ...   
2  INSTITUCION EDUCATIVA GIMNASIO PEDAGOGICO SAMP...  ...   
3  INSTITUCION EDUCATIVA GIMNASIO PEDAGOGICO SAMP...  ...   
4  INSTITUCION EDUCATIVA GIMNASIO PEDAGOGICO SAMP...  ...   

   estu_depto_presentacion estu_depto_reside  estu_mcpio_presentacion  \
0                    SUCRE             SUCRE                SINCELEJO   
1                    SUCRE             SUCRE                SINCELEJO   
2                    SUCRE             SUCRE                SINCELEJO   
3                    SUCRE             SUCRE                SINCELEJO   
4                    SUCRE             SUCRE                SINCELEJO   

   estu_mcpio_reside  punt_c_naturales punt_global punt_ingles  \
0            SAMPUÉS                39         202        39.0   
1            SAMPUÉS                42         232        52.0   
2            SAMPUÉS                44         206        32.0   
3            SAMPUÉS                33         177        31.0   
4            SAMPUÉS                35         186        45.0   

  punt_lectura_critica punt_matematicas punt_sociales_ciudadanas  
0                   43               36                       44  
1                   56               49                       37  
2                   47               36                       41  
3                   37               41                       32  
4                   42               41                       28  

[5 rows x 27 columns]

In [64]:
df_saber11_icfes.tail()

periodo cole_area_ubicacion  cole_cod_dane_establecimiento  \
11774    20242               RURAL                   270678000580   
11775    20242               RURAL                   270702000120   
11776    20242              URBANO                   170717000016   
11777    20242              URBANO                   170215000977   
11778    20242              URBANO                   170204000053   

       cole_cod_dane_sede cole_depto_ubicacion cole_jornada  \
11774        270678000580                SUCRE       MAÑANA   
11775        270702000120                SUCRE       MAÑANA   
11776        170717000016                SUCRE       MAÑANA   
11777        170215000977                SUCRE        TARDE   
11778        170204000053                SUCRE       MAÑANA   

      cole_mcpio_ubicacion cole_naturaleza      cole_nombre_establecimiento  \
11774              GALERAS         OFICIAL          INST EDUC PUERTO FRANCO   
11775  SAN JUAN DE BETULIA         OFICIAL                   I.E.  SABANETA   
11776            SAN PEDRO         OFICIAL       INST EDUC SAN PEDRO CLAVER   
11777              COROZAL         OFICIAL  I.E. NORMAL SUPERIOR DE COROZAL   
11778               COLOSÓ         OFICIAL              I.E. VICTOR ZUBIRIA   

                                       cole_nombre_sede  ...  \
11774                           INST EDUC PUERTO FRANCO  ...   
11775                    I.E. SABANETA - SEDE PRINCIPAL  ...   
11776                        INST EDUC SAN PEDRO CLAVER  ...   
11777  I.E. NORMAL SUPERIOR DE COROZAL - SEDE PRINCIPAL  ...   
11778              I.E. VICTOR ZUBIRIA - SEDE PRINCIPAL  ...   

       estu_depto_presentacion estu_depto_reside  estu_mcpio_presentacion  \
11774                    SUCRE             SUCRE                  GALERAS   
11775                    SUCRE             SUCRE                  COROZAL   
11776                    SUCRE             SUCRE                SAN PEDRO   
11777                    SUCRE             SUCRE                  COROZAL   
11778                    SUCRE             SUCRE                SINCELEJO   

         estu_mcpio_reside  punt_c_naturales punt_global punt_ingles  \
11774              GALERAS                45         206        26.0   
11775  SAN JUAN DE BETULIA                49         229        52.0   
11776            SAN PEDRO                49         229        52.0   
11777              COROZAL                58         279        54.0   
11778               COLOSÓ                61         328        57.0   

      punt_lectura_critica punt_matematicas punt_sociales_ciudadanas  
11774                   41               39                       45  
11775                   48               39                       45  
11776                   54               45                       33  
11777                   59               56                       51  
11778                   63               68                       73  

[5 rows x 27 columns]

In [65]:
df_saber11_icfes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11779 entries, 0 to 11778
Data columns (total 27 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   periodo                        11779 non-null  int64  
 1   cole_area_ubicacion            11779 non-null  object 
 2   cole_cod_dane_establecimiento  11779 non-null  int64  
 3   cole_cod_dane_sede             11779 non-null  int64  
 4   cole_depto_ubicacion           11779 non-null  object 
 5   cole_jornada                   11779 non-null  object 
 6   cole_mcpio_ubicacion           11779 non-null  object 
 7   cole_naturaleza                11779 non-null  object 
 8   cole_nombre_establecimiento    11779 non-null  object 
 9   cole_nombre_sede               11779 non-null  object 
 10  desemp_c_naturales             11779 non-null  int64  
 11  desemp_ingles                  11696 non-null  object 
 12  desemp_lectura_critica         11779 non-null 

## 4. Calidad del Dataset

### 4.1 Intituciones Educativas

In [66]:
# Cantidad de datos nulos
df_institucionesE.isnull().sum()

__id                      0
nombre_establecimiento    0
codigo_dane_sede          0
nombre_sede               0
zona                      0
dtype: int64

In [67]:
# Porcentaje de datos nulos
print('Porcentaje datos nulos')
df_institucionesE.isnull().mean()*100

Porcentaje datos nulos


__id                      0.0
nombre_establecimiento    0.0
codigo_dane_sede          0.0
nombre_sede               0.0
zona                      0.0
dtype: float64

In [68]:
# Porcentaje de datos no nulos
print('Porcentaje datos no nulos')
(1-df_institucionesE.isnull().mean())*100

Porcentaje datos no nulos


__id                      100.0
nombre_establecimiento    100.0
codigo_dane_sede          100.0
nombre_sede               100.0
zona                      100.0
dtype: float64

In [69]:
total_datos_institucionesE=df_institucionesE.shape[0]*df_institucionesE.shape[1]
total_nulos_institucionesE=df_institucionesE.isnull().sum().sum()
print('Porcentaje datos nulo en el dataset',(total_nulos_institucionesE/total_datos_institucionesE)*100)

Porcentaje datos nulo en el dataset 0.0


In [70]:
print('Porcentaje completitud del dataset', (1-total_nulos_institucionesE/total_datos_institucionesE)*100)

Porcentaje completitud del dataset 100.0


### 4.2 Conectividad

In [71]:
# Cantidad de datos nulos
df_conectividad.isnull().sum()

__id                           0
nombre_departamento            0
nombre_municipio               0
pdet                           0
nombre_centro_poblado          0
nombre_sede_educativa          0
fecha_inicio_de_operaci_n    296
fin_de_operaci_n_inicial     296
estado                         0
dtype: int64

In [72]:
# Porcentaje de datos nulos
print('Porcentaje datos nulos')
df_conectividad.isnull().mean()*100

Porcentaje datos nulos


__id                          0.000000
nombre_departamento           0.000000
nombre_municipio              0.000000
pdet                          0.000000
nombre_centro_poblado         0.000000
nombre_sede_educativa         0.000000
fecha_inicio_de_operaci_n    62.315789
fin_de_operaci_n_inicial     62.315789
estado                        0.000000
dtype: float64

In [73]:
# Porcentaje de datos no nulos
print('Porcentaje datos no nulos')
(1-df_conectividad.isnull().mean())*100

Porcentaje datos no nulos


__id                         100.000000
nombre_departamento          100.000000
nombre_municipio             100.000000
pdet                         100.000000
nombre_centro_poblado        100.000000
nombre_sede_educativa        100.000000
fecha_inicio_de_operaci_n     37.684211
fin_de_operaci_n_inicial      37.684211
estado                       100.000000
dtype: float64

In [74]:
total_datos_conectividad=df_conectividad.shape[0]*df_conectividad.shape[1]
total_nulos_conectividad=df_conectividad.isnull().sum().sum()
print('Porcentaje datos nulo en el dataset',(total_nulos_conectividad/total_datos_conectividad)*100)

Porcentaje datos nulo en el dataset 13.84795321637427


In [75]:
print('Porcentaje completitud del dataset', (1-total_nulos_conectividad/total_datos_conectividad)*100)

Porcentaje completitud del dataset 86.15204678362574


### 4.3 Pruebas Saber 11

#### 4.3.1 Dataset de Datos Abiertos

In [76]:
# Cantidad de datos nulos
df_saber11.isnull().sum()

periodo                              0
estu_tipodocumento                   0
estu_consecutivo                     0
cole_area_ubicacion                 23
cole_bilingue                     5867
cole_calendario                     13
cole_caracter                      540
cole_cod_dane_establecimiento       23
cole_cod_dane_sede                   3
cole_cod_depto_ubicacion             2
cole_cod_mcpio_ubicacion             2
cole_codigo_icfes                  191
cole_depto_ubicacion                 2
cole_genero                          2
cole_jornada                         2
cole_mcpio_ubicacion                 2
cole_naturaleza                      2
cole_nombre_establecimiento          2
cole_nombre_sede                     2
cole_sede_principal                  2
estu_cod_depto_presentacion          0
estu_cod_mcpio_presentacion          0
estu_cod_reside_depto              309
estu_cod_reside_mcpio              309
estu_depto_presentacion              0
estu_depto_reside                  309
estu_estadoinvestigacion             0
estu_estudiante                      0
estu_fechanacimiento                 0
estu_genero                         23
estu_mcpio_presentacion              0
estu_mcpio_reside                  309
estu_nacionalidad                    0
estu_pais_reside                     0
estu_privado_libertad                0
fami_cuartoshogar                  698
fami_educacionmadre               1224
fami_educacionpadre               1220
fami_estratovivienda              1166
fami_personashogar                 668
fami_tieneautomovil                716
fami_tienecomputador               894
fami_tieneinternet                1044
fami_tienelavadora                 689
desemp_ingles                        9
punt_ingles                          0
punt_matematicas                     0
punt_sociales_ciudadanas         21536
punt_c_naturales                 21536
punt_lectura_critica             21536
punt_global                      21536
dtype: int64

In [77]:
df_saber11_sucre.isnull().sum()

periodo                            0
estu_tipodocumento                 0
estu_consecutivo                   0
cole_area_ubicacion                0
cole_bilingue                    129
cole_calendario                    0
cole_caracter                      1
cole_cod_dane_establecimiento      0
cole_cod_dane_sede                 0
cole_cod_depto_ubicacion           0
cole_cod_mcpio_ubicacion           0
cole_codigo_icfes                  2
cole_depto_ubicacion               0
cole_genero                        0
cole_jornada                       0
cole_mcpio_ubicacion               0
cole_naturaleza                    0
cole_nombre_establecimiento        0
cole_nombre_sede                   0
cole_sede_principal                0
estu_cod_depto_presentacion        0
estu_cod_mcpio_presentacion        0
estu_cod_reside_depto              5
estu_cod_reside_mcpio              5
estu_depto_presentacion            0
estu_depto_reside                  5
estu_estadoinvestigacion           0
estu_estudiante                    0
estu_fechanacimiento               0
estu_genero                        0
estu_mcpio_presentacion            0
estu_mcpio_reside                  5
estu_nacionalidad                  0
estu_pais_reside                   0
estu_privado_libertad              0
fami_cuartoshogar                 17
fami_educacionmadre               25
fami_educacionpadre               24
fami_estratovivienda              20
fami_personashogar                14
fami_tieneautomovil               16
fami_tienecomputador              17
fami_tieneinternet                21
fami_tienelavadora                16
desemp_ingles                      0
punt_ingles                        0
punt_matematicas                   0
punt_sociales_ciudadanas         379
punt_c_naturales                 379
punt_lectura_critica             379
punt_global                      379
dtype: int64

In [78]:
# Porcentaje de datos nulos
print('Porcentaje datos nulos')
df_saber11.isnull().mean()*100

Porcentaje datos nulos


periodo                           0.000
estu_tipodocumento                0.000
estu_consecutivo                  0.000
cole_area_ubicacion               0.046
cole_bilingue                    11.734
cole_calendario                   0.026
cole_caracter                     1.080
cole_cod_dane_establecimiento     0.046
cole_cod_dane_sede                0.006
cole_cod_depto_ubicacion          0.004
cole_cod_mcpio_ubicacion          0.004
cole_codigo_icfes                 0.382
cole_depto_ubicacion              0.004
cole_genero                       0.004
cole_jornada                      0.004
cole_mcpio_ubicacion              0.004
cole_naturaleza                   0.004
cole_nombre_establecimiento       0.004
cole_nombre_sede                  0.004
cole_sede_principal               0.004
estu_cod_depto_presentacion       0.000
estu_cod_mcpio_presentacion       0.000
estu_cod_reside_depto             0.618
estu_cod_reside_mcpio             0.618
estu_depto_presentacion           0.000
estu_depto_reside                 0.618
estu_estadoinvestigacion          0.000
estu_estudiante                   0.000
estu_fechanacimiento              0.000
estu_genero                       0.046
estu_mcpio_presentacion           0.000
estu_mcpio_reside                 0.618
estu_nacionalidad                 0.000
estu_pais_reside                  0.000
estu_privado_libertad             0.000
fami_cuartoshogar                 1.396
fami_educacionmadre               2.448
fami_educacionpadre               2.440
fami_estratovivienda              2.332
fami_personashogar                1.336
fami_tieneautomovil               1.432
fami_tienecomputador              1.788
fami_tieneinternet                2.088
fami_tienelavadora                1.378
desemp_ingles                     0.018
punt_ingles                       0.000
punt_matematicas                  0.000
punt_sociales_ciudadanas         43.072
punt_c_naturales                 43.072
punt_lectura_critica             43.072
punt_global                      43.072
dtype: float64

In [79]:
# Porcentaje de datos nulos
print('Porcentaje datos nulos')
df_saber11_sucre.isnull().mean()*100

Porcentaje datos nulos


periodo                           0.000000
estu_tipodocumento                0.000000
estu_consecutivo                  0.000000
cole_area_ubicacion               0.000000
cole_bilingue                    14.222712
cole_calendario                   0.000000
cole_caracter                     0.110254
cole_cod_dane_establecimiento     0.000000
cole_cod_dane_sede                0.000000
cole_cod_depto_ubicacion          0.000000
cole_cod_mcpio_ubicacion          0.000000
cole_codigo_icfes                 0.220507
cole_depto_ubicacion              0.000000
cole_genero                       0.000000
cole_jornada                      0.000000
cole_mcpio_ubicacion              0.000000
cole_naturaleza                   0.000000
cole_nombre_establecimiento       0.000000
cole_nombre_sede                  0.000000
cole_sede_principal               0.000000
estu_cod_depto_presentacion       0.000000
estu_cod_mcpio_presentacion       0.000000
estu_cod_reside_depto             0.551268
estu_cod_reside_mcpio             0.551268
estu_depto_presentacion           0.000000
estu_depto_reside                 0.551268
estu_estadoinvestigacion          0.000000
estu_estudiante                   0.000000
estu_fechanacimiento              0.000000
estu_genero                       0.000000
estu_mcpio_presentacion           0.000000
estu_mcpio_reside                 0.551268
estu_nacionalidad                 0.000000
estu_pais_reside                  0.000000
estu_privado_libertad             0.000000
fami_cuartoshogar                 1.874311
fami_educacionmadre               2.756340
fami_educacionpadre               2.646086
fami_estratovivienda              2.205072
fami_personashogar                1.543550
fami_tieneautomovil               1.764057
fami_tienecomputador              1.874311
fami_tieneinternet                2.315325
fami_tienelavadora                1.764057
desemp_ingles                     0.000000
punt_ingles                       0.000000
punt_matematicas                  0.000000
punt_sociales_ciudadanas         41.786108
punt_c_naturales                 41.786108
punt_lectura_critica             41.786108
punt_global                      41.786108
dtype: float64

In [80]:
# Porcentaje de datos no nulos
print('Porcentaje datos no nulos')
(1-df_saber11.isnull().mean())*100

Porcentaje datos no nulos


periodo                          100.000
estu_tipodocumento               100.000
estu_consecutivo                 100.000
cole_area_ubicacion               99.954
cole_bilingue                     88.266
cole_calendario                   99.974
cole_caracter                     98.920
cole_cod_dane_establecimiento     99.954
cole_cod_dane_sede                99.994
cole_cod_depto_ubicacion          99.996
cole_cod_mcpio_ubicacion          99.996
cole_codigo_icfes                 99.618
cole_depto_ubicacion              99.996
cole_genero                       99.996
cole_jornada                      99.996
cole_mcpio_ubicacion              99.996
cole_naturaleza                   99.996
cole_nombre_establecimiento       99.996
cole_nombre_sede                  99.996
cole_sede_principal               99.996
estu_cod_depto_presentacion      100.000
estu_cod_mcpio_presentacion      100.000
estu_cod_reside_depto             99.382
estu_cod_reside_mcpio             99.382
estu_depto_presentacion          100.000
estu_depto_reside                 99.382
estu_estadoinvestigacion         100.000
estu_estudiante                  100.000
estu_fechanacimiento             100.000
estu_genero                       99.954
estu_mcpio_presentacion          100.000
estu_mcpio_reside                 99.382
estu_nacionalidad                100.000
estu_pais_reside                 100.000
estu_privado_libertad            100.000
fami_cuartoshogar                 98.604
fami_educacionmadre               97.552
fami_educacionpadre               97.560
fami_estratovivienda              97.668
fami_personashogar                98.664
fami_tieneautomovil               98.568
fami_tienecomputador              98.212
fami_tieneinternet                97.912
fami_tienelavadora                98.622
desemp_ingles                     99.982
punt_ingles                      100.000
punt_matematicas                 100.000
punt_sociales_ciudadanas          56.928
punt_c_naturales                  56.928
punt_lectura_critica              56.928
punt_global                       56.928
dtype: float64

In [81]:
# Porcentaje de datos no nulos
print('Porcentaje datos no nulos')
(1-df_saber11_sucre.isnull().mean())*100

Porcentaje datos no nulos


periodo                          100.000000
estu_tipodocumento               100.000000
estu_consecutivo                 100.000000
cole_area_ubicacion              100.000000
cole_bilingue                     85.777288
cole_calendario                  100.000000
cole_caracter                     99.889746
cole_cod_dane_establecimiento    100.000000
cole_cod_dane_sede               100.000000
cole_cod_depto_ubicacion         100.000000
cole_cod_mcpio_ubicacion         100.000000
cole_codigo_icfes                 99.779493
cole_depto_ubicacion             100.000000
cole_genero                      100.000000
cole_jornada                     100.000000
cole_mcpio_ubicacion             100.000000
cole_naturaleza                  100.000000
cole_nombre_establecimiento      100.000000
cole_nombre_sede                 100.000000
cole_sede_principal              100.000000
estu_cod_depto_presentacion      100.000000
estu_cod_mcpio_presentacion      100.000000
estu_cod_reside_depto             99.448732
estu_cod_reside_mcpio             99.448732
estu_depto_presentacion          100.000000
estu_depto_reside                 99.448732
estu_estadoinvestigacion         100.000000
estu_estudiante                  100.000000
estu_fechanacimiento             100.000000
estu_genero                      100.000000
estu_mcpio_presentacion          100.000000
estu_mcpio_reside                 99.448732
estu_nacionalidad                100.000000
estu_pais_reside                 100.000000
estu_privado_libertad            100.000000
fami_cuartoshogar                 98.125689
fami_educacionmadre               97.243660
fami_educacionpadre               97.353914
fami_estratovivienda              97.794928
fami_personashogar                98.456450
fami_tieneautomovil               98.235943
fami_tienecomputador              98.125689
fami_tieneinternet                97.684675
fami_tienelavadora                98.235943
desemp_ingles                    100.000000
punt_ingles                      100.000000
punt_matematicas                 100.000000
punt_sociales_ciudadanas          58.213892
punt_c_naturales                  58.213892
punt_lectura_critica              58.213892
punt_global                       58.213892
dtype: float64

In [82]:
total_datos_saber11=df_saber11.shape[0]*df_saber11.shape[1]
total_nulos_saber11=df_saber11.isnull().sum().sum()
print('Porcentaje datos nulo en el dataset',(total_nulos_saber11/total_datos_saber11)*100)

Porcentaje datos nulo en el dataset 4.016117647058824


In [83]:
total_datos_saber11_sucre=df_saber11_sucre.shape[0]*df_saber11_sucre.shape[1]
total_nulos_saber11_sucre=df_saber11_sucre.isnull().sum().sum()
print('Porcentaje datos nulo en el dataset',(total_nulos_saber11_sucre/total_datos_saber11_sucre)*100)

Porcentaje datos nulo en el dataset 3.97345266662343


In [84]:
print('Porcentaje completitud del dataset', (1-total_nulos_saber11/total_datos_saber11)*100)

Porcentaje completitud del dataset 95.98388235294118


In [85]:
print('Porcentaje completitud del dataset', (1-total_nulos_saber11_sucre/total_datos_saber11_sucre)*100)

Porcentaje completitud del dataset 96.02654733337657


#### 4.3.2 Dataset de Repositorio ICFES

In [86]:
# Cantidad de datos nulos
df_saber11_icfes.isnull().sum()

periodo                             0
cole_area_ubicacion                 0
cole_cod_dane_establecimiento       0
cole_cod_dane_sede                  0
cole_depto_ubicacion                0
cole_jornada                        0
cole_mcpio_ubicacion                0
cole_naturaleza                     0
cole_nombre_establecimiento         0
cole_nombre_sede                    0
desemp_c_naturales                  0
desemp_ingles                      83
desemp_lectura_critica              0
desemp_matematicas                  0
desemp_sociales_ciudadanas          0
estu_dedicacioninternet          1449
estu_dedicacionlecturadiaria     1427
estu_depto_presentacion             0
estu_depto_reside                   7
estu_mcpio_presentacion             0
estu_mcpio_reside                   7
punt_c_naturales                    0
punt_global                         0
punt_ingles                        83
punt_lectura_critica                0
punt_matematicas                    0
punt_sociales_ciudadanas            0
dtype: int64

In [87]:
# Porcentaje de datos nulos
print('Porcentaje datos nulos')
df_saber11_icfes.isnull().mean()*100

Porcentaje datos nulos


periodo                           0.000000
cole_area_ubicacion               0.000000
cole_cod_dane_establecimiento     0.000000
cole_cod_dane_sede                0.000000
cole_depto_ubicacion              0.000000
cole_jornada                      0.000000
cole_mcpio_ubicacion              0.000000
cole_naturaleza                   0.000000
cole_nombre_establecimiento       0.000000
cole_nombre_sede                  0.000000
desemp_c_naturales                0.000000
desemp_ingles                     0.704644
desemp_lectura_critica            0.000000
desemp_matematicas                0.000000
desemp_sociales_ciudadanas        0.000000
estu_dedicacioninternet          12.301554
estu_dedicacionlecturadiaria     12.114781
estu_depto_presentacion           0.000000
estu_depto_reside                 0.059428
estu_mcpio_presentacion           0.000000
estu_mcpio_reside                 0.059428
punt_c_naturales                  0.000000
punt_global                       0.000000
punt_ingles                       0.704644
punt_lectura_critica              0.000000
punt_matematicas                  0.000000
punt_sociales_ciudadanas          0.000000
dtype: float64

In [88]:
# Porcentaje de datos no nulos
print('Porcentaje datos no nulos')
(1-df_saber11_icfes.isnull().mean())*100

Porcentaje datos no nulos


periodo                          100.000000
cole_area_ubicacion              100.000000
cole_cod_dane_establecimiento    100.000000
cole_cod_dane_sede               100.000000
cole_depto_ubicacion             100.000000
cole_jornada                     100.000000
cole_mcpio_ubicacion             100.000000
cole_naturaleza                  100.000000
cole_nombre_establecimiento      100.000000
cole_nombre_sede                 100.000000
desemp_c_naturales               100.000000
desemp_ingles                     99.295356
desemp_lectura_critica           100.000000
desemp_matematicas               100.000000
desemp_sociales_ciudadanas       100.000000
estu_dedicacioninternet           87.698446
estu_dedicacionlecturadiaria      87.885219
estu_depto_presentacion          100.000000
estu_depto_reside                 99.940572
estu_mcpio_presentacion          100.000000
estu_mcpio_reside                 99.940572
punt_c_naturales                 100.000000
punt_global                      100.000000
punt_ingles                       99.295356
punt_lectura_critica             100.000000
punt_matematicas                 100.000000
punt_sociales_ciudadanas         100.000000
dtype: float64

In [89]:
total_datos_saber11_icfes=df_saber11_icfes.shape[0]*df_saber11_icfes.shape[1]
total_nulos_df_saber11_icfes=df_saber11_icfes.isnull().sum().sum()
print('Porcentaje datos nulo en el dataset',(total_nulos_df_saber11_icfes/total_datos_saber11_icfes)*100)

Porcentaje datos nulo en el dataset 0.9609065725883792


In [90]:
print('Porcentaje completitud del dataset', (1-total_nulos_df_saber11_icfes/total_datos_saber11_icfes)*100)

Porcentaje completitud del dataset 99.03909342741161


## **Exploración de los Datos y Negocio**

1. **Dataset de Instituciones Educativas**
        •	Total de registros: 800
        •	Columnas: 5
        •	Porcentaje de datos nulos: 0%
        •	Completitud: 100%
  **Comentario:** Este dataset está completamente limpio, sin valores nulos. No necesita tratamiento adicional respecto a calidad de datos.

  **Nota:** El dataset de instituciones educativas públicas del departamento de Sucre fue considerado como fuente complementaria para contextualizar el análisis. Aunque no es estrictamente necesario para responder la pregunta de investigación, que se enfoca en la relación entre conectividad institucional y puntajes en las Pruebas Saber 11, su inclusión permite identificar el universo total de sedes educativas oficiales, y con ello analizar aspectos como la cobertura de conectividad y la proporción de instituciones con estudiantes evaluados. Por esta razón, su uso queda sujeto al enfoque específico que se adopte en el análisis final.

2. **Dataset de Conectividad**
        •	Total de registros: 475
        •	Columnas: 9
        •	Variables con nulos:
              o	fecha_inicio_de_operaci_n: 296 valores nulos (≈62.3%)
              o	fin_de_operaci_n_inicial: 296 valores nulos (≈62.3%)
        •	Porcentaje de datos nulos global: 13.85%
        •	Completitud del dataset: 86.15%
  **Comentario:** Las fechas de operación presentan un alto porcentaje de valores nulos (más del 60%). Actualmente estamos evaluando si estas variables son esenciales para el análisis. En caso de que resulten clave para responder la pregunta de investigación, se considerarán opciones como la imputación o un tratamiento específico de los valores faltantes. De lo contrario, podrían ser excluidas para preservar la consistencia del análisis.

3. **Dataset Pruebas Saber 11 - Datos Abiertos (50,000 registros)**
        •	Porcentaje de completitud global: ≈95.98%
        •	Columnas con más nulos:
              o	cole_bilingue: 11.7%
              o	fami_educacionmadre: 2.4%
              o	punt_sociales_ciudadanas, punt_lectura_critica, punt_c_naturales, punt_global: 43.1% cada uno
  **Comentario:** El mayor problema está en las variables de puntajes, con más del 40% faltante. Puede deberse a estudiantes exentos o a inconsistencias en el reporte. Esto podría afectar el análisis académico, por lo tanto, se considera excluir esos registros y trabajar solo con puntajes completos.

4. **Dataset Pruebas Saber 11 - SUCRE (907 registros filtrados)**
        •	Porcentaje de completitud global: ≈96.03%
        •	Columnas con nulos destacadas:
            o	cole_bilingue: 14.2%
            o	punt_sociales_ciudadanas, punt_c_naturales, punt_lectura_critica, punt_global: 41.8%
  **Comentario:** A pesar de estar más enfocado, el comportamiento de los nulos es similar al dataset completo (sin filtrar por el departamento de Sucre).  Podríamos validar si los puntajes faltantes corresponden a un año o sede específicos.

5. **Dataset Pruebas Saber 11 - Repositorio ICFES (11,779 registros)**
        •	Porcentaje de datos nulos global: 0.96%
        •	Completitud: ≈99.04%
        •	Variables con nulos:
              o	estu_dedicacioninternet: 12.3%
              o	estu_dedicacionlecturadiaria: 12.1%
              o	punt_ingles: 0.7%
  **Comentario:** Este dataset está bien estructurado, y la mayoría de las columnas tienen una completitud muy alta. Solo dos variables de hábitos tienen nulos, que pueden imputarse o analizarse por separado.



  **Resumen General:**

  | **Dataset**                     | **% Nulos** | **Completitud** | **Observación clave**                          |
  |--------------------------------|-------------|------------------|------------------------------------------------|
  | Instituciones Educativas       | 0%          | 100%             | Sin nulos                                      |
  | Conectividad                   | 13.85%      | 86.15%           | Fechas con alta tasa de nulos                  |
  | Saber 11 (Datos Abiertos)            | 4.01%       | 95.98%           | Puntajes con muchos nulos                      |
  | Saber 11 Sucre                | 3.97%       | 96.03%           | Patrón similar al dataset general              |
  | Saber 11 ICFES (Repositorio)   | 0.96%       | 99.04%           | Buena calidad; hábitos con algunos nulos       |